extensions/WSLExtension/Services/WslServicesMediator.cs

```diff
@@ -0,0 +1,208 @@
+        //
+        // Note: the distribution that's set up as the default, will contain (default) next to it. But for our purposes
+        // we don't need to read that part. We only need to first word of the space separated line. Distribution
+        // names cannot have spaces so we don't need to worry about that either.
+        reader.ReadLine();
+        while (reader.ReadLine() is { } line)
+        {
+            var spaceSeparatedArr = line.Split(" ");
+            distributions.Add(spaceSeparatedArr[FirstIndex]);
+        }
+
+        return distributions;
+    }
+
+    /// <inheritdoc cref="IWslServicesMediator.IsDistributionRunning"/>
+    public bool IsDistributionRunning(string distributionName)
+    {
+        return GetAllNamesOfRunningDistributions().Contains(distributionName);
+    }
+
+    /// <inheritdoc cref="IWslServicesMediator.GetAllRegisteredDistributions"/>
+    /// <remarks>
+    /// Method enumerates through the WSL registry location subkey location
+    /// to retrieve information about each registered distribution.
+    /// </remarks>
+    public List<WslRegisteredDistribution> GetAllRegisteredDistributions()
+    {
+        var distributions = new List<WslRegisteredDistribution>();
+        var linuxSubSystemKey = CurrentUser.OpenSubKey(WslRegistryLocation, false);
+
+        if (linuxSubSystemKey == null)
+        {
+            return new();
+        }
+
+        var defaultDistribution = linuxSubSystemKey.GetValue(DefaultDistributionRegistryName) as string;
+
+        foreach (var subKeyName in linuxSubSystemKey.GetSubKeyNames())
+        {
+            var subKey = linuxSubSystemKey.OpenSubKey(subKeyName);
+
+            if (subKey == null)
+            {
+                continue;
+            }
+
+            var distribution = BuildDistributionInfoFromRegistry(subKey);
+            if (string.IsNullOrEmpty(distribution.Name))
+            {
+                // distribution doesn't have a name. This would happen only if the users registry info
+                // was messed up. WSL would likely not function properly either in these cases.
+                continue;
+            }
+
+            // the last part of the registry subkey is the registered guid of the wsl distribution.
+            var distributionGuid = subKey.Name.Split('\\').LastOrDefault() ?? string.Empty;
+            if (!string.IsNullOrEmpty(defaultDistribution) &&
+                defaultDistribution.Equals(distributionGuid, StringComparison.OrdinalIgnoreCase))
+            {
+                distribution.IsDefaultDistribution = true;
+            }
+
+            distributions.Add(distribution);
+        }
+
+        return distributions;
+    }
+
+    private WslRegisteredDistribution BuildDistributionInfoFromRegistry(RegistryKey registryKey)
+    {
+        var regDistributionName = registryKey.GetValue(DistributionRegistryName) as string ?? string.Empty;
+        var subkeyName = registryKey.Name.Split('\\').LastOrDefault();

```

Comment:
nit; "subkeyName" is confusing. It's registryKey name relative to the parent, but it sounds like it's one of the subkeys of registryKey.

Reply:
Fair I updated the name to be `lastSegmentOfRegistryKeyName`. Hope that makes it clearer. Thanks